### PR TITLE
Enhance error logging in `archiveHelper.hpp` for improved diagnostics

### PR DESCRIPTION
**Pull request by GabrielEValenzuela**
|Related issue|
|---|
|#24326|

## Objective
This PR enhances the error logging capabilities within the `archiveHelper.hpp` file. The goal is to provide more detailed and informative error messages that will aid in diagnostics and troubleshooting, improving the overall robustness and maintainability of the system.

## Description
- During routine maintenance and debugging sessions, it was observed that the current error logging in `archiveHelper.hpp` lacks sufficient detail, which hampers effective problem-solving. The existing logs do not provide enough context about the errors, making it challenging to pinpoint issues quickly and accurately.
- To address this, we have implemented a series of improvements in how errors are logged within this component.

### Static Analysis
- The updated error logging code underwent a thorough static analysis to ensure that it adheres to coding standards and best practices, particularly in relation to safe string handling and memory management.

#### Unit Testing
N/A

------------------------------

### CODE REVIEW

**Review by sebasfalcone — CHANGES_REQUESTED** [review]
Some aesthetic changes to improve the logging

File: src/shared_modules/utils/archiveHelper.hpp [hunk 1]
```diff
@@ -112,7 +113,8 @@ namespace Utils
 
             if (retVal != ARCHIVE_OK)
             {
-                throw std::runtime_error("Couldn't open file");
+                auto errMsg = archive_error_string(archiveRead.get());
+                throw std::runtime_error("Couldn't open file during decompress. Error: " + std::string(errMsg ? errMsg : "Unknown error"));
```
[the remark this answers is not in the document]
Comment (GabrielEValenzuela) [reply src/shared_modules/utils/archiveHelper.hpp]: Done. Thanks!

File: src/shared_modules/utils/archiveHelper.hpp [hunk 2]
```diff
@@ -125,7 +127,8 @@ namespace Utils
 
                 if (retVal != ARCHIVE_OK)
                 {
-                    throw std::runtime_error("Couldn't read file");
+                    auto errMsg = archive_error_string(archiveRead.get());
+                    throw std::runtime_error("Couldn't open file during reading next header. Error: " + std::string(errMsg ? errMsg : "Unknown error"));
```
[the remark this answers is not in the document]
Comment (GabrielEValenzuela) [reply src/shared_modules/utils/archiveHelper.hpp]: Done. Thanks!

File: src/shared_modules/utils/archiveHelper.hpp [hunk 3]
```diff
@@ -65,7 +65,8 @@ namespace Utils
 
                 if (retVal != ARCHIVE_OK)
                 {
-                    throw std::runtime_error("Couldn't read file");
+                    auto errMsg = archive_error_string(archiveRead);
+                    throw std::runtime_error("Couldn't open file during copy data. Error: " + std::string(errMsg ? errMsg : "Unknown error"));
```
[the remark this answers is not in the document]
Comment (GabrielEValenzuela) [reply src/shared_modules/utils/archiveHelper.hpp]: Done. Thanks!

**Review by sebasfalcone — DISMISSED** [review]
LGTM!

**Review by sebasfalcone — APPROVED** [review]

**Pull request merged**